src/debugAdapter.ts

```diff
@@ -2,136 +2,112 @@
  * Copyright (C) Microsoft Corporation. All rights reserved.
  *--------------------------------------------------------*/
 
-import fs = require("fs");
-import net = require("net");
-import os = require("os");
-import path = require("path");
+import {connect, Socket } from "net";
+import { DebugAdapter, Event, DebugProtocolMessage, EventEmitter } from "vscode";
 import { Logger } from "./logging";
-import utils = require("./utils");
-
-// NOTE: The purpose of this file is to serve as a bridge between
-// VS Code's debug adapter client (which communicates via stdio) and
-// PowerShell Editor Services' debug service (which communicates via
-// named pipes or a network protocol).  It is purely a naive data
-// relay between the two transports.
-
-const logBasePath = path.resolve(__dirname, "../../logs");
-
-const debugAdapterLogWriter =
-    fs.createWriteStream(
-        path.resolve(
-            logBasePath,
-            "DebugAdapter.log"));
-
-// Pause the stdin buffer until we're connected to the
-// debug server
-process.stdin.pause();
-
-const debugSessionFilePath = utils.getDebugSessionFilePath();
-debugAdapterLogWriter.write("Session file path: " + debugSessionFilePath + ", pid: " + process.pid + " \r\n");
-
-function startDebugging() {
-    // Read the details of the current session to learn
-    // the connection details for the debug service
-    const sessionDetails = utils.readSessionFile(debugSessionFilePath);
-
-    // TODO: store session details into an in-memory store that can be shared between
-    // the debug adapter and client extension
-    // and then clean up the session details file.
-
-    // Establish connection before setting up the session
-    debugAdapterLogWriter.write("Connecting to pipe: " + sessionDetails.debugServicePipeName + "\r\n");
-
-    let isConnected = false;
-    const debugServiceSocket = net.connect(sessionDetails.debugServicePipeName);
-
-    // Write any errors to the log file
-    debugServiceSocket.on(
-        "error",
-        (e) => {
-            debugAdapterLogWriter.write("Socket ERROR: " + e + "\r\n");
-            debugAdapterLogWriter.close();
-            debugServiceSocket.destroy();
-            process.exit(0);
+
+export class NamedPipeDebugAdapter implements DebugAdapter {
+    private static readonly TWO_CRLF = '\r\n\r\n';
+    private static readonly HEADER_LINESEPARATOR = /\r?\n/;	// allow for non-RFC 2822 conforming line separators
+    private static readonly HEADER_FIELDSEPARATOR = /: */;
+
+    private _rawData = Buffer.allocUnsafe(0);
+    private _contentLength = -1;
+    private _isConnected: boolean = false;
+    private _debugMessageQueue: DebugProtocolMessage[] = [];
+
+    private _debugServiceSocket: Socket;
```

Comment:
Could these be added to the constructor for injection-initialisation?

Reply:
decided against this because I wanted the `_` prefix.